src/20241119_AaveV3EthereumLido_GHOListingOnLidoPool/GHODirectMinter.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IPool, DataTypes} from 'aave-v3-origin/contracts/interfaces/IPool.sol';
+import {OwnableUpgradeable} from 'openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol';
+import {Initializable} from 'openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol';
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {SafeERC20} from 'solidity-utils/contracts/oz-common/SafeERC20.sol';
+import {IGhoToken} from '../interfaces/IGhoToken.sol';
+
+/**
+ * @title GHODirectMinter
+ * @notice Acts as a facilitator
+ * Instead of minting and burning on demand on each user interaction, the facilitator mints an amount upfront and makes it available via supply.
+ * @author BGD Labs @bgdlabs
+ */
+contract GHODirectMinter is Initializable, OwnableUpgradeable {
+  using SafeERC20 for IERC20;
+  IPool public immutable POOL;
+  address public immutable COLLECTOR;
+  IGhoToken public immutable GHO;
+
+  constructor(IPool pool, address collector, address gho) {
+    POOL = pool;
+    COLLECTOR = collector;
+    GHO = IGhoToken(gho);
+    _disableInitializers();
+  }
+
+  function initialize(address owner) external virtual initializer {
+    __Ownable_init(owner);
+  }
+
+  /**
+   * @dev mints GHO and supplies it to the pool
+   */
+  function mintAndSupply(uint256 amount) external onlyOwner {
+    GHO.mint(address(this), amount);
+    IERC20(address(GHO)).forceApprove(address(POOL), amount);
+    POOL.supply(address(GHO), amount, address(this), 0);
+  }
+
+  /**
+   * @dev withdraws GHO from the pool and burns it
+   */
+  function withdrawAndBurn(uint256 amount) external onlyOwner {
+    uint256 amountWithdrawn = POOL.withdraw(address(GHO), amount, address(this));
+
+    GHO.burn(amountWithdrawn);
+  }
+
+  function transferExcessToTreasury() external {
+    (uint256 capacity, ) = GHO.getFacilitatorBucket(address(this));
+    DataTypes.ReserveDataLegacy memory reserveData = POOL.getReserveData(address(GHO));
```

Comment:
aToken can be immutable as well, no point to query every time

Reply:
the reason i query it here is that otherwise i would need to deploy the minter on the proposal, which makes testing a bit madness (as i would need to somehow extract the added address).

If you insist i can do it, but to me feel cleaner this way and 10k gas saved on a rather rare operation probably is not so important. We can still do it in an upgrade.